src/components/logs/Context.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+    createContext,
+    useContext,
+    useEffect,
+    useCallback,
+    useState,
+    useRef,
+} from 'react';
+import { BaseComponentProps, ViewLogs_Line } from 'types';
+import { DEFAULT_POLLING_INTERVAL, RPCS } from 'services/supabase';
+import { hasLength, incrementInterval, timeoutCleanUp } from 'utils/misc-utils';
+import { useClient } from 'hooks/supabase-swr';
+
+interface Props extends BaseComponentProps {
+    token: string | null;
+    disableIntervalFetching?: boolean;
+    fetchAll?: boolean;
+}
+
+type LogsContextValue = {
+    reset: () => void;
+    logs: ViewLogs_Line[];
+    networkFailure: boolean;
+    stopped: boolean;
+};
+
+const MAX_EMPTY_CALLS = 90; // about 1.5 minute
+const MAX_INTERVAL = 1000; //  we don't want to wait too longer between pings
+const LogsContext = createContext<LogsContextValue | null>(null);
+
+const LogsContextProvider = ({
+    children,
+    token,
+    disableIntervalFetching,
+    fetchAll,
+}: Props) => {
+    const supabaseClient = useClient();
+
+    // Privates
+    const allLogs = useRef<ViewLogs_Line[]>([]);
+    const offset = useRef(0);
+    const timeoutRef = useRef<number | undefined>();
+    const interval = useRef(DEFAULT_POLLING_INTERVAL / 2);
+    const emptyResponses = useRef(
+        disableIntervalFetching ? MAX_EMPTY_CALLS : 0
+    );
```

Comment:
Need some stuff we don't _really_ need to make public and need refs so they update when we need 'em.